doc/source/user/misc.rst

```diff
@@ -56,159 +68,178 @@ and ``'ignore'`` for ``underflow``.  But this can be changed, and it can be
+- ``'log'``    : Record error in a Log object specified by :func:`~numpy.seterrcall`.
 
 These behaviors can be set for all kinds of errors or specific ones:
 
- - all       : apply to all numeric exceptions
- - invalid   : when NaNs are generated
- - divide    : divide by zero (for integers as well!)
- - overflow  : floating point overflows
- - underflow : floating point underflows
+- ``all``       : apply to all numeric exceptions
+- ``invalid``   : when NaNs are generated
+- ``divide``    : divide by zero (for integers as well!)
+- ``overflow``  : floating point overflows
+- ``underflow`` : floating point underflows
 
 Note that integer divide-by-zero is handled by the same machinery.
-These behaviors are set on a per-thread basis.
+These behaviors are set on a per-thread basis. Each thread in a multithreaded
+program can have its own error handling configuration. See :ref:`thread_safety`
+for more information about using NumPy in multithreaded contexts.
+
+The error handling mode can be configured using :func:`numpy.seterr`,
+:func:`numpy.geterr`, :func:`numpy.seterrcall`, and the :func:`numpy.errstate`
+context manager. See :doc:`/reference/routines.err` for full API documentation.
 
 Examples
 --------
 
 ::
 
- >>> oldsettings = np.seterr(all='warn')
- >>> np.zeros(5,dtype=np.float32)/0.
- Traceback (most recent call last):
- ...
- RuntimeWarning: invalid value encountered in divide
- >>> j = np.seterr(under='ignore')
- >>> np.array([1.e-100])**10
- array([0.])
- >>> j = np.seterr(invalid='raise')
- >>> np.sqrt(np.array([-1.]))
- Traceback (most recent call last):
- ...
- FloatingPointError: invalid value encountered in sqrt
- >>> def errorhandler(errstr, errflag):
- ...      print("saw stupid error!")
- >>> np.seterrcall(errorhandler)
- >>> j = np.seterr(all='call')
- >>> np.zeros(5, dtype=np.int32)/0
- saw stupid error!
- array([nan, nan, nan, nan, nan])
- >>> j = np.seterr(**oldsettings) # restore previous
- ...                              # error-handling settings
+   >>> oldsettings = np.seterr(all='warn')
+   >>> np.zeros(5,dtype=np.float32)/0.
+   Traceback (most recent call last):
+   ...
+   RuntimeWarning: invalid value encountered in divide
+
+::
+
+    >>> j = np.seterr(under='ignore')
+    >>> np.array([1.e-100])**10
+    array([0.])
+
+::
+
+   >>> j = np.seterr(invalid='raise')
+   >>> np.sqrt(np.array([-1.]))
+   Traceback (most recent call last):
+   ...
+   FloatingPointError: invalid value encountered in sqrt
+
+::
+
+   >>> def errorhandler(errstr, errflag):
+   ...      print("saw stupid error!")
+   >>> np.seterrcall(errorhandler)
+   >>> j = np.seterr(all='call')
+   >>> np.zeros(5, dtype=np.int32)/0
+   saw stupid error!
+   array([nan, nan, nan, nan, nan])
+
+::
+
+   >>> j = np.seterr(**oldsettings) # restore previous
+   ...                              # error-handling settings
 
 Interfacing to C
 ----------------
 Only a survey of the choices. Little detail on how each works.
```

Comment:
I think most of what is below here could/should use some pruning. But that might be another PR.